scrapy/core/downloader/webclient.py

```diff
@@ -7,17 +7,19 @@
 
 from scrapy.http import Headers
 from scrapy.utils.httpobj import urlparse_cached
+from scrapy.utils.python import to_bytes
 from scrapy.responsetypes import responsetypes
 
 
 def _parsed_url_args(parsed):
     path = urlunparse(('', '', parsed.path or '/', parsed.params, parsed.query, ''))
-    host = parsed.hostname
+    path = to_bytes(path)
+    host = to_bytes(parsed.hostname)
     port = parsed.port
-    scheme = parsed.scheme
-    netloc = parsed.netloc
+    scheme = to_bytes(parsed.scheme, encoding='ascii')
+    netloc = to_bytes(parsed.netloc)
     if port is None:
-        port = 443 if scheme == 'https' else 80
+        port = 443 if scheme == b'https' else 80
     return scheme, netloc, host, port, path
```

Comment:
Yes, indeed urls here come from Request.url, that passes them via ``w3lib.url.safe_url_string``, so they are ascii-only - I added a comment and specified ascii encoding in to_bytes.